electionleaflets/apps/leaflets/models.py

```diff
@@ -111,6 +114,36 @@ def get_party(self):
 
         return None
 
+    def attach_nuts_code(self) -> None:
+        """
+        Gets a NUTS1 code from YNR.
+
+        This is a little hacky: we might not have a ballot for a postcode at the
+        point we look up that postcode. But we want a NUTS1 code for every
+        uploaded leaflet.
+
+        Because of this, we pick a known UK-wide election date and look up
+        what the NUTS1 code of that ballot ID was. This in imperfect for
+        most geographies, but NUTS1 are big enough that we don't care about
+        (literal) edge cases.
+
+        """
+
+        params = {
+            "for_postcode": self.postcode,
+            "election_type": "parl",
+            "election_date": "2024-07-04",
+        }
+        ynr_helper = YNRAPIHelper()
+        try:
+            resp = ynr_helper.get("ballots", params=params)
+            self.nuts1 = resp["results"][0]["tags"].get("NUTS1", {}).get("key")
+        except Exception as e:
```

Comment:
if we're catching generic exception, I'd do:

```py
try:
    # generic thing that might fail in an unknown way
except (KeyboardInterrupt, SystemExit):
    raise
except Exception as err:
    # exception behaviour
    # maybe re-raise
```

otherwise things can get wierd

Reply:
I don't think that' needed...`Exception` inherits from `BaseException`, as does `KeyboardInterrupt` and `SystemExit`. So I'm basically not catching those last two anyway. See https://pywheel.com/python-exceptions/